proposals/0179-stricter-verification.md

```diff
@@ -28,19 +28,39 @@ The new syscall instruction introduced in SIMD-0178 permits the
 differentiation of internal and external calls, so we can introduce new 
 verification rules to prevent any unwarranted behavior.
 
+Another motivitation is to have the ability to treat functions as
+self-contained compilation units, allowing a hybrid virtual machine that works 
+with both JIT compilation and an interpreter, depending on the trade off 
+between performance and compilation time on a per function basis.
+
 ## New Terminology
 
 None.
 
 ## Detailed Design
 
 The following must go into effect if and only if a program indicates the SBF 
-version XX or higher in its ELF header e_flags field, according to the 
+version `0x3` or higher in its ELF header e_flags field, according to the 
 specification of SIMD-0161.
 
+### Restrict functions' first instruction
+
+Since SIMD-0166, SBPF functions can modify the stack pointer using the 
+`add64 r10, imm` (opcode `0x07`) with the restriction that `r10` is the 
+destination register and the source is an immediate value.
+
+The instruction `add64 r10, imm` must be used as a marker for a function 
```

Comment:
This should also mention that a function can not have another `add64 r10, imm` as that would count as the start of the next function / split the function in two.

Reply:
`r10` is restored upon return automatically (it is on the VMs shadow stack: https://github.com/anza-xyz/sbpf/blob/main/src/interpreter.rs#L579). Thus, there is no need to counter act the stack allocation with a second stack bump in the opposite direction I think.